### PR TITLE
Enable extra warnings (-Wextra)

### PR DESCRIPTION
**Pull request by rigtorp**
Enable extra warnings and fix those warnings.

### CODE REVIEW

**Comment by rigtorp** [comment]
@tmontgomery What do you think about this patch?

File: config/suppress.xml [hunk 1]
```diff
@@ -4,4 +4,5 @@
         "http://www.puppycrawl.com/dtds/suppressions_1_0.dtd">
 <suppressions>
     <suppress files=".*generated.*" checks="."/>
+    <suppress files="CppGenerator.java" checks="MethodLength"/>
```
Comment (tmontgomery) [inline config/suppress.xml]: This is not something we want to suppress.

Comment (rigtorp) [inline config/suppress.xml]: I don't want to split up the constructor codegen function for no good reason. Should I do something like this:
```
void generateConstructor() {
  generatePart1();
  generatePart2();
}
```

Comment (mjpt777) [inline config/suppress.xml]: Yes it is best to break up large methods into logical blocks so they can be reasoned about.

**Comment by ksergey** [comment]
a looking forward 80c53b6 in next release

**Comment by rigtorp** [comment]
@tmontgomery @mjpt777 I've updated this PR to ignore -Wtype-limits in wrapForEncode instead of the more complicated change i previously had.

This PR is a superset of #421

**Comment by rigtorp** [comment]
@tmontgomery @mjpt777 Any thoughts on this?

**Comment by tmontgomery** [comment]
Sorry this took so long. Thanks!

**Pull request merged**